File: Secondary-School-(St.-Joseph's-Convent)/Main Projects/StudentsGrade.pas
Program StudentsGrade;

{This program inputs the name of a student and the mark they received, determines the grade, and outputs the student's name and their grade}
{Created by : Arianna St.John}
{Date : 13/01/2022}

Var
    Name : String;
    Mark : Real;
    Grade : Char;
Begin
    Write ('Enter the name and mark of a student: ');
    Readln (Name, Mark);
    If Mark >= 80 
    Then
        Grade := 'A'
    Else
    If (Mark >= 65) and (Mark < 80) 
    Then
        Grade := 'B'
    Else
    If (Mark >= 50) and (Mark < 65) 
    Then
        Grade := 'C'
    Else
    If (Mark >= 35) and (Mark < 50) 
    Then
        Grade := 'D'
    Else
    If Mark < 35 
    Then
        Grade := 'F';
    Writeln ('The student''s name is: ', Name);
    Writeln ('The student''s grade is: ', Grade);
End.

File: College-(Sir-Arthur-Lewis-Community-College)/Small Exercises/C-&C++/studentsGrade.cpp
 /*Program studentsGrade

This program prompts the user to create a file and input students' names and grades into it
Created by : Arianna St.John
Date : 09/04/24*/ 

#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main(){

    string fname = " ";
    string lname = " ";
    char grade = ' ';
    int con = 1;

    ofstream outFile;
    outFile.open("grades.txt", fstream::app);

    if (! outFile){
  	    cerr << "Error! Unable to open file." << endl;
    }
    else {
   	    cout << "File successfully created!" << endl;
    }

    while (con != 0){
        cout << "\n";
        cout << "Enter the student's first name: " << "\t\t";
        cin >> fname;
        cout << "\n";
        cout << "Enter the student's last name:  " << "\t\t";
        cin >> lname;
        cout << "\n";
        cout << "Enter the student's grade (A, B, C, D, E, F):" << "\t";
        cin >> grade;
        cout << "\n";
        cout << "Enter 0 to stop:" << "\t\t\t\t\t\t\t";
        cin >> con;
        outFile << fname << "\t\t" << lname << "\t\t" << grade << endl;
        outFile << "\n";
    }

    cout << "\n";
    cout << "File successfully written to!" << endl;
    outFile.close();

return 0;}